mods/env_sounds/init.lua

```diff
@@ -0,0 +1,97 @@
+-- Parameters
+
+local radius = 8 -- Water node search radius around player
+
+-- End of parameters
+
+
+local handles = {}
+local river_source_sounds = minetest.settings:get_bool("river_source_sounds")
+
+
+-- Update sound for player
+
+local function update_sound(player)
+	local player_name = player:get_player_name()
+	-- Search for water nodes in radius around player
+	local ppos = player:get_pos()
+	local areamin = vector.subtract(ppos, radius)
+	local areamax = vector.add(ppos, radius)
+	local wpos, num
+	if river_source_sounds then
+		wpos, num = minetest.find_nodes_in_area(
+			areamin,
+			areamax,
+			{
+				"default:water_flowing",
+				"default:river_water_source",
+				"default:river_water_flowing"
+			}
+		)
+	else
+		wpos, num = minetest.find_nodes_in_area(
+			areamin,
+			areamax,
+			{
+				"default:water_flowing",
+				"default:river_water_flowing"
+			}
+		)
+	end
+	-- Total number of waters in radius
+	local waters = #wpos
+	if waters == 0 then
+		return
+	end
+
+	-- Find average position of water positions
+	local wposav = vector.new()
+	for i, pos in ipairs(wpos) do
+		wposav = vector.add(wposav, pos)
```

Comment:
This creates many new tables and might be inefficient. In-place adding is probably better:
```suggestion
		wposav.x = wposav.x + pos.x
		wposav.y = wposav.y + pos.y
		wposav.z = wposav.z + pos.z
```

Reply:
Not sure about this, can't judge.

Reply:
Will wait for core dev input.

Reply:
Can't hurt to do this IMO.

Reply:
I agree, but will it help performance?

Reply:
Afaik, memory allocation can be slow. Speed tests with `core.get_us_time()` are possible.

Reply:
Well, i'll do this, as it seems sort of supported by sfan5.

Reply:
Well, i'll do this.